### PR TITLE
Output gap conductivity as a material property

### LINKED ISSUE #9226: Output gap conductivity as a material property
### Description of the enhancement or error report
Output the gap conductivity calculation in GapConductance.C as a material property.

### Rationale for the enhancement or information for reproducing the error
The thermal conductivity of the gap is required as an input in another material that is under development.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

New material property available for output and usage in other materials.

### PR DESCRIPTION
**Pull request by gambka**
Closes #9226.

### CODE REVIEW

**Comment by dschwen** [comment]
Just a question: I do not understand this conceptually. The gap is an _unmeshed_ region. So how can you have a material property describe some gap property? The material property can only live where the gap is _not_. What is meaning of the property at the points where it is defined?

**Comment by gambka** [comment]
We already output the gap conductance as a material property that is defined on the slave surface in thermal contact and zero elsewhere.  Conceptually this is done so we can visualize what the value of the gap conductance is or use the value elsewhere.  However, in my case I need the thermal conductivity of the gas within the gap for use in another material property and thus I followed the same procedure as the gap conductance.

File: modules/heat_conduction/include/materials/GapConductance.h [hunk 1]
```diff
@@ -88,6 +88,7 @@ class GapConductance : public Material
   const VariableValue & _gap_temp_value;
   MaterialProperty<Real> & _gap_conductance;
   MaterialProperty<Real> & _gap_conductance_dT;
+  MaterialProperty<Real> & _gap_conductivity_property;
```
Comment (permcody) [inline modules/heat_conduction/include/materials/GapConductance.h]: Minor: Do you really want the suffix `_propery` on there? We don't typically do this.

File: modules/heat_conduction/src/materials/GapConductance.C [hunk 2]
```diff
@@ -124,6 +124,7 @@ GapConductance::GapConductance(const InputParameters & parameters)
     _gap_temp_value(_quadrature ? _zero : coupledValue("gap_temp")),
     _gap_conductance(declareProperty<Real>("gap_conductance" + _appended_property_name)),
     _gap_conductance_dT(declareProperty<Real>("gap_conductance" + _appended_property_name + "_dT")),
+    _gap_conductivity_property(declareProperty<Real>("gap_conductivity")),
```
Comment (permcody) [inline modules/heat_conduction/src/materials/GapConductance.C]: Would you mind either adding a new test or updating an existing test so we can keep this small change working?
Thanks

**Review by permcody — CHANGES_REQUESTED** [review]

File: modules/heat_conduction/include/materials/GapConductance.h (hunk 1 above)
Comment (gambka) [inline modules/heat_conduction/include/materials/GapConductance.h]: I would use _gap_conductivity as the member variable name but this name is already used in this class. I could use _gap_thermal_conductivity as the name instead.

File: modules/heat_conduction/src/materials/GapConductance.C (hunk 2 above)
Comment (gambka) [inline modules/heat_conduction/src/materials/GapConductance.C]: I'll make a test.

File: modules/heat_conduction/include/materials/GapConductance.h (hunk 1 above)
Comment (permcody) [inline modules/heat_conduction/include/materials/GapConductance.h]: Oh! Well you might consider changing the other member. It's just there as a bulk property right? Do what you think is best.

**Review by permcody — APPROVED** [review]

**Pull request merged**